pkg/usagestats/reporter.go

```diff
@@ -55,7 +107,96 @@ func (r *Reporter) running(ctx context.Context) error {
 
 	level.Info(r.logger).Log("msg", "usage stats reporter initialized", "cluster_id", seed.UID)
 
-	// TODO Periodically send usage report.
+	// Find when to send the next report. We want all instances of the same Mimir cluster computing the same value.
+	nextReportAt := nextReport(r.reportSendInterval, seed.CreatedAt, time.Now())
+
+	ticker := time.NewTicker(r.reportCheckInterval)
+	defer ticker.Stop()
+
+	for {
+		select {
+		case <-ticker.C:
+			if time.Now().Before(nextReportAt) {
+				continue
+			}
+
+			// If the send is failing since a long time and the report is falling behind,
+			// we'll skip this one and try to send the next one.
+			if time.Since(nextReportAt) >= r.reportSendInterval {
+				nextReportAt = nextReport(r.reportSendInterval, seed.CreatedAt, time.Now())
+				level.Info(r.logger).Log("msg", "failed to send anonymous usage stats report for too long, skipping to next report", "next_report_at", nextReportAt.String())
+				continue
+			}
+
+			level.Debug(r.logger).Log("msg", "sending anonymous usage stats report")
+			if err := r.sendReport(ctx, buildReport(seed, nextReportAt, r.reportSendInterval)); err != nil {
+				level.Info(r.logger).Log("msg", "failed to send anonymous usage stats report", "err", err)
+
+				// We'll try at the next check interval.
+				continue
+			}
+
+			nextReportAt = nextReport(r.reportSendInterval, seed.CreatedAt, time.Now())
+		case <-ctx.Done():
+			if err := ctx.Err(); !errors.Is(err, context.Canceled) {
+				return err
+			}
+			return nil
+		}
+	}
+}
+
+// sendReport sends the report to the stats server.
+func (r *Reporter) sendReport(ctx context.Context, report Report) (returnErr error) {
+	startTime := time.Now()
+	r.requestsTotal.Inc()
+
+	defer func() {
+		r.requestsLatency.Observe(time.Since(startTime).Seconds())
+		if returnErr != nil {
+			r.requestsFailedTotal.Inc()
+		}
+	}()
+
+	data, err := json.Marshal(report)
+	if err != nil {
+		return errors.Wrap(err, "failed to marshal the report")
+	}
+	req, err := http.NewRequest(http.MethodPost, r.serverURL, bytes.NewBuffer(data))
```

Comment:
Nit, as we don't expect anyone to write there:
```suggestion
	req, err := http.NewRequest(http.MethodPost, r.serverURL, bytes.NewReader(data))
```